### PR TITLE
PLA-6443: update sample app.

### PR DESCRIPTION
**Pull request by cgranthamtyro**
Adds ConnectionProvider implementation (POSConnectionProvider) which implements some of the functionality intended for the POS back-end for the purpose of demonstrating how the connection secret is retrieved for a registered reader id.
Adds environment configuration so that the app can distinguish between production and production sandbox environments.
Simple UI enables payments and refunds to be tested.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -37,9 +38,10 @@ let package = Package(
             ]
     ),
     // Hosted frameworks
-    .binaryTarget(name: "TyroTapToPaySDK",
-                  url: "https://tap-to-pay.connect.tyro.com/tyro/0.2.0/TyroTapToPaySDK.xcframework.zip",
-                  checksum: "65014d1ba962dbf5c04342d9eeed4b31815a188beee74af1117a47befba76577"),
+//    .binaryTarget(name: "TyroTapToPaySDK",
```
Comment (sanj-tyro) [inline Package.swift:41]: todo: upload the xcframework to s3

File: SampleApp/SampleApp.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -0,0 +1,697 @@
...
+/* Begin PBXFrameworksBuildPhase section */
+		8B76E0CD2B5782500017B455 /* Frameworks */ = {
+			isa = PBXFrameworksBuildPhase;
+			buildActionMask = 2147483647;
+			files = (
+				BDDDB1FE2BA95F8A004385DB /* TyroTapToPaySDKPackage in Frameworks */,
+				BD0A73232B9F502200659F82 /* TyroTapToPaySDKPackage in Frameworks */,
+				BDFDEE622B8F03760034DA1F /* TyroTapToPaySDKPackage in Frameworks */,
+				BD9CEA762B9ABB2B00928203 /* OpenSSL in Frameworks */,
+				BD1633782B9ACB6800183BE7 /* netfox in Frameworks */,
```
Comment (wrutkowski-tyro) [inline SampleApp/SampleApp.xcodeproj/project.pbxproj:78]: do you want to include netfox in the sample app?

File: SampleApp/SampleApp/Supporting Files/Info.plist [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	<string>${ATTESTATION_REFRESH_INTERVAL}</string>
+	<key>KEYLOADING_CA_CERT</key>
+	<string>$(KEYLOADING_CA_CERT)</string>
+	<key>KEYLOADING_HOST</key>
+	<string>$(KEYLOADING_HOST)</string>
+	<key>KEYLOADING_HOST_CERT_PINNING</key>
+	<string>$(KEYLOADING_HOST_CERT_PINNING)</string>
+	<key>NSAppTransportSecurity</key>
+	<dict>
+		<key>NSAllowsArbitraryLoads</key>
```
Comment (wrutkowski-tyro) [inline SampleApp/SampleApp/Supporting Files/Info.plist:23]: is this needed?

File: SampleApp/SampleApp/View/PaymentsView.swift [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+      self.error = "\(error)"
+    }
+  }
+
+  func startRefund() {
+    guard isAAmountValid(amount) else {
+      error = "Wrong amount format"
+      return
+    }
+
```
Comment (wrutkowski-tyro) [inline SampleApp/SampleApp/View/PaymentsView.swift:70]: we are missing refund implementation

File: SampleApp/SampleApp/View/PaymentsView.swift [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+          HStack(spacing: 10) {
+            ProgressView()
+            Text(processInProgress)
+          }
+          Spacer()
+        }
+        .padding()
+      }
+      VStack {
+        //        TextField("Amount", value: $viewModel.amount, format: .currency(code: .))
```
Comment (wrutkowski-tyro) [inline SampleApp/SampleApp/View/PaymentsView.swift:110]: should we remove this line?

File: SampleApp/SampleApp/View/PaymentsView.swift [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+      inputFocus = .amountTextField
+      await viewModel.connect()
+    }
+  }
+}
+
+#Preview {
+
+
+
```
Comment (wrutkowski-tyro) [inline SampleApp/SampleApp/View/PaymentsView.swift:160]: we could remove those empty lines

File: Package.swift (hunk 1 above)
Comment (cgranthamtyro) [reply Package.swift]: removed 👍🏼

File: SampleApp/SampleApp/Supporting Files/Info.plist (hunk 3 above)
Comment (cgranthamtyro) [reply SampleApp/SampleApp/Supporting Files/Info.plist]: Also removed 👍🏼

File: SampleApp/SampleApp/View/PaymentsView.swift (hunk 4 above)
Comment (sanj-tyro) [reply SampleApp/SampleApp/View/PaymentsView.swift]: added in refund flow

**Review by wrutkowski-tyro — APPROVED** [review]

**Review by sanj-tyro — APPROVED** [review]

**Pull request merged**